services/quiz.js

```diff
@@ -40,8 +40,16 @@ const quizService = {
       throw createForbiddenError()
     }
 
-    for (let field in updateData) {
-      quiz[field] = updateData[field]
+    const { settings } = updateData
+
+    if (settings) {
+      for (let props in settings) {
+        quiz.settings[props] = settings[props]
+      }
+    } else {
+      for (let field in updateData) {
+        quiz[field] = updateData[field]
+      }

```

Comment:
what if update data will contain setting and other properties? As I see it will update only settings.

previous update didn`t work for setting, because it will rewrite the hole settings object I guess

Reply:
> what if updateData will contain settings and other properties? As I see it will update only settings.
> 
> Previous update didn`t work for settings? because it will rewrite the whole settings object I guess

Changed) 

Quiz before edit:
<img width="419" alt="Знімок екрана 2023-12-01 о 10 42 19" src="https://github.com/ita-social-projects/SpaceToStudy-BackEnd/assets/17857767/5c16e062-8b4f-40e7-a49e-d3b4c0d1cddd">


Request body:
<img width="419" alt="Знімок екрана 2023-12-01 о 10 43 04" src="https://github.com/ita-social-projects/SpaceToStudy-BackEnd/assets/17857767/a50d6358-7cdd-4641-a71b-ca13fc85d064">

Quiz after change: 
<img width="419" alt="Знімок екрана 2023-12-01 о 10 43 50" src="https://github.com/ita-social-projects/SpaceToStudy-BackEnd/assets/17857767/1722b7e6-fc0d-45e0-89f5-75c254945e68">